### PR TITLE
feat: Added mail channel implementation

### PR DESCRIPTION
**Pull request by sonimadhuri**
This feature will allow users to send notifications via mail.

API Enforcements: 
- Requires specifying Mail channel in notification channels and need to define `toMail()` method that returns mail data.
- Notifiable mail recipients can be specified with `getMailRecipients()`

Handling of mails is delegated to [nest-mailman](https://github.com/squareboat/nest-mailman)

### CODE REVIEW

File: lib/generics/notifiable.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export interface Notifiable {
   getFcmTokens(): string | string[];
```
Comment (vinayak25) [inline lib/generics/notifiable.ts]: Need to make both the properties as optional.

File: lib/interfaces/options.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 
+import { MailmanOptions } from '@squareboat/nest-mailman/dist/interfaces';
+
 export interface NotificationOptions {
   channels: {
     fcm: {
       credentialsPath: string
-    }
+    },
+    mail: MailmanOptions
```
Comment (vinayak25) [inline lib/interfaces/options.ts]: Need to make both the properties "fcm" and "mail" as optional

**Review by vinayak25 — CHANGES_REQUESTED** [review]

**Comment by sonimadhuri** [comment]
@vinayak25 PR updated. Please check.

**Comment by vinayak25** [comment]
Merge Done! Congratulations @sonimadhuri

**Pull request merged**